R/read-gs.R

```diff
@@ -15,11 +16,200 @@ read_gs <- function(path, verbose = FALSE){
   gs <- read_csv(path, trim_ws = FALSE) |>
     #check format
     check_data_format()
+  #add source attribute
+  attr(gs, "source") <- "Gradescope"
+  
+  gs
+}
+
+
+#'Read in 1 or more files from Grading Platform
+#'
+#'This function will read in input data and combine in the appropriate manner.
+#'@param grades_path Is the path to the csv containing graded assignments.
+#'@param other_file_paths Is a list containing any other filepaths desired to be read in. Each name should be a designated type of file. For example, "lateness" and "roster" and the value should be the filepath.
+#'@param source Determines how to read the grade file. By default, it is auto so that Gradebook can determine what format the file is in. Alternatively, users can specify "Gradescope" or "Canvas" to force certain interpretations.
+#'@return A dataframe of grades. Later support for other files will be returned as a list as well.
+#'@importFrom readr read_csv
+#'@export
+read_files <- function(grades_path, 
+                       other_file_paths = list(), 
+                       source = "auto"){
+  #first read in all of the files
+  
+  #begininng with the grades csv
+  
+  tryCatch({
+    grades <- readr::read_csv(grades_path, trim_ws = FALSE)
+    },
+    error = function(e){
```

Comment:
I don't think this check is necessary. I'm sure `read_csv()` has it's own error message regarding improper paths. Best to leave it to the function that's actually doing the thing.